clients/iOS/Grapevine/Grapevine/Event.swift

```diff
@@ -33,16 +31,37 @@ class Event: NSObject, Mappable {
         
     }
     
+    
+    // Default map function doesn't work for dates for some reason hence this
+    func dateMap(dict: [String: AnyObject]){
+        // Events will always have a start time and therefore a date
+        
+        let dateFromJson = NSDate(timeIntervalSince1970: NSTimeInterval(dict["start_time"] as! String)!)
+        self.startTime = EventTime()
+        self.startTime.setAll(dateFromJson)
+        
+        if (dict["date"] != nil){
+            self.dateNS = NSDate(timeIntervalSince1970: NSTimeInterval(dict["date"] as! String)!)
+            self.date = CVDate(date: self.dateNS)
```

Comment:
not that I could find. this time stuff caused me a bunch of trouble. spend days on it.